### ISSUE TITLE
showChallenge() method throws NetworkOnMainThreadException

### ISSUE THREAD

**Issue by sanusi87** [ayltai/Android-Lib-reCAPTCHA#5]
I tried to reload the captcha once a user entered a wrong answer, I use this code:

````java
ReCaptcha reCaptcha = (ReCaptcha)findViewById(R.id.recaptcha);
reCaptcha.showChallenge(SITE_KEY);
````

then it throws NetworkOnMainThreadException. I tried running it inside AsyncTask, then an error notice appears saying that showChallenge() must be called from the UI thread, currently inferred thread is worker...

any suggestion?

**Comment by ayltai** [comment]
In your case you need to use `showChallengeAsync`. Try invoking `showChallengeAsync` on your UI thread.

For historical reason, `showChallenge` is synchronous and it touches UI components so it is only intended for use in SDKs prior to Honeycomb.

**Comment by ayltai** [comment]
This issue is resolved in version 2.0.0.
